Sources/NIOCore/IPAddresses.swift

```diff
@@ -0,0 +1,586 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2021 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+
+import Darwin
+import CNIOLinux
+import CoreFoundation
+
+
+/// `Error` that may be thrown if we fail to create a `IPAddress`
+public enum IPAddressError: Error {
```

Comment:
New top-level types need to be prefixed with `NIO`. As there are quite a few new top-level types in this proposal, we may want to nest them all within `public enum IPAddress` (which will have to become `public enum NIOIPAddress`), as that avoids us needing to namespace them all.